applications/nrf_desktop/doc/hid_forward.rst

```diff
@@ -93,33 +98,27 @@ Forwarding HID input reports
+In the current implementation, the :ref:`nrf_desktop_usb_state` supports either the HID boot keyboard or the HID boot mouse reports, because the application does not support assigning HID boot protocol code separately for each USB HID instance.
 
 The :c:func:`hogp_read` callback is called when HID input report is received from the connected peripheral.
-The received HID input report data is converted to ``hid_report_event``.
+The received HID input report data is added to the HID report queue and converted to a :c:struct:`hid_report_event`.
 
-``hid_report_event`` is submitted and then the HID-class USB device configured by :ref:`nrf_desktop_usb_state` forwards it to the host.
-When a HID report is sent to the host by the HID-class USB device, the |hid_forward| receives a ``hid_report_sent_event`` with the identifier of this device.
+The :c:struct:`hid_report_event` is submitted by the HID report queue and the HID-class USB device configured by :ref:`nrf_desktop_usb_state` forwards it to the host.
+When a HID report is sent to the host by the HID-class USB instance, the |hid_forward| receives a :c:struct:`hid_report_sent_event` with the identifier of the instance.
+The |hid_forward| notifies the HID report queue to ensure proper HID report flow.
 
 Enqueuing incoming HID input reports
 ------------------------------------
 
-The |hid_forward| forwards only one HID input report to the HID-class USB device at a time.
-Another HID input report may be received from a peripheral connected over Bluetooth before the previous one was sent.
-In that case, ``hid_report_event`` is enqueued and submitted later.
-Up to the number of reports specified in :ref:`CONFIG_DESKTOP_HID_FORWARD_MAX_ENQUEUED_REPORTS <config_desktop_app_options>` Kconfig option can be enqueued at a time for each report type and for each HID subscriber (HID-class USB device).
-If there is not enough space to enqueue a new event, the module drops the oldest enqueued event (of the same type) that was enqueued for a given HID subscriber.
-
-Upon receiving the ``hid_report_sent_event``, the |hid_forward| submits the ``hid_report_event`` enqueued for the peripheral that is associated with the HID-class USB device.
-The enqueued report to be sent is chosen by the |hid_forward| in the round-robin fashion.
-The report of the next type will be sent if available.
-If not available, the next report type will be checked until a report is found or there is no report in any of the queues.
-If there is no ``hid_report_event`` in the queue, the module waits for receiving data from peripherals.
+USB HID class instances limit the maximum number of HID input reports that can be handled simultaneously.
+The number of HID input reports received from HID peripherals connected over Bluetooth LE may exceed the limit.
```

Comment:
Bluetooth® LE

Reply:
@peknis should we update it here?

Reply:
We update only for the 1st instance and its already done. Hence, we need not update here.